### PR TITLE
feat(admin-next) discounts list page

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- added `/discounts` page

**Note**
- discount detail view and create/edit will be covered in the following PRs

### CODE REVIEW

**Comment by srindom** [comment]
- If you filter by dynamic type on this the app crashes.
- Would expect there to be filters for created_at too

File: packages/admin-next/dashboard/src/components/table/table-cells/discount/redemption-cell/redemption-cell.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export const RedemptionHeader = () => {
+  const { t } = useTranslation()
+
+  return (
+    <div className="flex h-full w-full items-center justify-end">
+      <span>{t("fields.totalRedemptions")}</span>
+    </div>
+  )
+}
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/components/table/table-cells/discount/redemption-cell/redemption-cell.tsx]: nit: use `text-right` for both of these. The header wraps incorrectly on small screen currently. Also might want to add nowrap to the header.

File: packages/admin-next/dashboard/public/locales/en-US/translation.json [hunk 2]
```diff
@@ -355,6 +362,11 @@
     "sales_channels": "Sales Channels",
     "status": "Status",
     "code": "Code",
+    "value": "Value",
+    "disabled": "Disabled",
+    "dynamic": "dynamic",
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/public/locales/en-US/translation.json]: Capitalize

File: packages/admin-next/dashboard/public/locales/en-US/translation.json [hunk 3]
```diff
@@ -355,6 +362,11 @@
...
+    "normal": "normal",
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/public/locales/en-US/translation.json]: Capitalize

File: packages/admin-next/dashboard/public/locales/en-US/translation.json [hunk 4]
```diff
@@ -187,7 +187,14 @@
...
   "discounts": {
-    "domain": "Discounts"
+    "domain": "Discounts",
+    "deleteWarning": "You are about to delete the discount {{title}}. This action cannot be undone.",
+    "discountStatus": {
+      "scheduled": "scheduled",
+      "expired": "expired",
+      "active": "active",
+      "disabled": "disabled"
+    }
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/public/locales/en-US/translation.json]: Capitalize

File: packages/admin-next/dashboard/src/hooks/table/columns/use-discount-table-columns.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+      columnHelper.accessor("rule.value", {
+        header: () => <ValueHeader />,
+        cell: ({ row }) => (
+          <ValueCell
+            rule={row.original.rule}
+            currencyCode={row.original.regions[0].currency_code}
+          />
+        ),
+      }),
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/hooks/table/columns/use-discount-table-columns.tsx]: This throws an error for dynamic discounts, need to add a check to prevent it

File: packages/admin-next/dashboard/src/routes/discounts/list/components/discount-list-table/discount-list-table.tsx [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+        columns={columns}
+        count={count}
+        pageSize={PAGE_SIZE}
+        filters={filters}
+        search
+        pagination
+        isLoading={isLoading}
+        queryObject={raw}
+        navigateTo={(row) => `${row.original.id}`}
+      />
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/routes/discounts/list/components/discount-list-table/discount-list-table.tsx]: suggestion: Add `order`. If it's not supported by the endpoint then I would add it, but also fine to do in a separate PR.

**Review by kasperkristensen — COMMENTED** [review]
LGTM just a couple minor things that should be addressed before we merge

File: packages/admin-next/dashboard/src/routes/discounts/list/components/discount-list-table/discount-list-table.tsx (hunk 6 above)
Comment (fPolic) [reply packages/admin-next/dashboard/src/routes/discounts/list/components/discount-list-table/discount-list-table.tsx]: added ✅

File: packages/admin-next/dashboard/src/hooks/table/filters/use-discount-table-filters.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import { useTranslation } from "react-i18next"
+import { Filter } from "../../../components/table/data-table"
+
+export const useDiscountTableFilters = () => {
+  const { t } = useTranslation()
+
+  let filters: Filter[] = []
```
Comment (kasperkristensen) [inline packages/admin-next/dashboard/src/hooks/table/filters/use-discount-table-filters.tsx]: todo: Add `created_at` and `updated_at` filters

**Review by kasperkristensen — APPROVED** [review]
LGTM! Just need to add date filters but otherwise it's good to go

**Comment by olivermrbl** [comment]
Don't we need date filter support on the backend too?
